uwblueprint/jump-math
# feat: add admin mutation

## Notion ticket link

[Add Admin Mutation](https://www.notion.so/uwblueprintexecs/Add-Admin-Mutation-f95fcd0c4e3b41e2ad2b4e6ac283092a)



## Implementation description
* AddAdminModal.tsx: call the `AddUser` mutation in `onSubmit()` (which runs when the add admin modal form is submitted) to add a new admin user to the database, and added an error alert to the modal that displays when either (a) one or more of the fields is blank upon submitting, or (b) a request error occurs
* AdminConfirmationMessage.tsx, MessageContainer.tsx: updated these two components to be consistent with the Figma designs for the confirmation message styling + button positioning
* typography.ts: added `mobileSubtitle1` as a new text style (as shown [here](https://www.figma.com/file/Uia7C5xwqDlkLnrXMyt5hn/Jump-Math-Design-System?node-id=5%3A2&t=DFdbqB6TzBDnptdU-0)) to be used with the error alerts
* UserTypes.ts: defined new types for the add admin mutation request/response
* UserMutations.ts: created a new `AddUser` mutation to be used by the add admin modal
* Renamed `GetUsersByRole.ts` to `UserQueries.ts` to be consistent with the [folder structure in the Rowan House Society repo](https://github.com/uwblueprint/rowan-house/tree/main/frontend/src/APIClients)
* misc. lint fixes in other files


## Steps to test
1. Click the "+ Add Admin" button on the admin landing page to bring up the add admin modal
2. Fill out the form. Try submitting 
    - with one or more fields blank
    - with an improperly-formatted email
    - with a non-jump math email if the `Yes` radio button is selected
    - with something else that will cause an API request error (you might need to bork the `onSubmit()` code slightly to mock this)
to verify the different failure scenarios as specified in the [Figma design](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=484%3A17091&t=gBJNG7Y2JuEjzspd-0) 
3. Once the form has been submitted successfully (with all inputs valid), connect to MongoDB and verify that a new admin user has been added under the `test.users` collection, with credentials matching those that were inputted


## What should reviewers focus on?
<img width="1321" alt="Screen Shot 2022-11-28 at 1 43 28 AM" src="https://user-images.githubusercontent.com/68651393/204211112-95ddaf6a-ba66-4f26-a83b-b85c0c3f493a.png">
<img width="1321" alt="Screen Shot 2022-11-28 at 1 41 43 AM" src="https://user-images.githubusercontent.com/68651393/204210797-e4ffae04-77a2-4c7f-918a-412c453b9268.png">

[Figma](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=484%3A17091&t=Z3gXSSguA2LMnKB6-0)

## Checklist
